pages/apis/graphql/cookbooks/rules.md

```diff
@@ -0,0 +1,95 @@
+# Rules
```

Comment:
I couldn't figure out how to get this added to the graphql nav, which is autogenerated 🤔

Reply:
@l-suzuki , have you tried [running through this process](https://github.com/buildkite/docs/blob/main/README.md#updating-graphql-api-docs)?
This runs off the latest/current version of the GraphQL API.

Reply:
Hi @l-suzuki , as is mentioned at the top of the [`data/nav_graphql.yml` file](https://github.com/buildkite/docs/blob/main/data/nav_graphql.yml), have you tried editing the [`scripts/graphql_api_content/nav_data.rb` file](https://github.com/buildkite/docs/blob/main/scripts/graphql_api_content/nav_data.rb) to add rules entry to it?

Would that work?

Reply:
Sorry @l-suzuki , it was through a broken link reported by the docs build linter that I now know what you mean here.

No, these pages are not auto-generated, although they are a part of the `nav_data.rb` file I linked to above. Therefore, I'll add this to that nav file and clarify this in the recently added and revamped (from original material) 'contributors guide'.

Reply:
Ah! Then one has to run `bundle exec rake graphql:generate` to regenerate the `nav_graphql.yml` file 🤦

Reply:
And then add the equivalent `nav_data.rb` entry to the [`/spec/scripts/graphql_api_content/nav_data_spec.rb` file](https://github.com/buildkite/docs/blob/main/spec/scripts/graphql_api_content/nav_data_spec.rb) to get the builds to finally pass.